src/providers/one_inch.rs

```diff
@@ -165,4 +180,58 @@ impl ConversionProvider for OneInchProvider {
 
         Ok(response)
     }
+
+    async fn build_approve_tx(
+        &self,
+        params: ConvertApproveQueryParams,
+    ) -> RpcResult<ConvertApproveResponseBody> {
+        let chain_id = crypto::disassemble_caip10(&params.from)?.1;
+        let (_, dst_chain_id, dst_address) = crypto::disassemble_caip10(&params.to)?;
+
+        // Check if from and to chain ids are different
+        // 1inch provider does not support cross-chain swaps
+        if dst_chain_id != chain_id {
+            return Err(RpcError::InvalidParameter(
+                "from and to chain ids are different in a single chain swap".into(),
```

Comment:
What makes it a single chain swap?

```suggestion
                "Swapping between chains is currently not supported".into(),
```

Reply:
We will use one inch for only single swap transactions, so this should be better filtered in the handler function first and this case should not occur at the provider itself. I'll make a follow up to move this check at the provider choice at the handler. Thanks for raising this!